### ISSUE TITLE
p:repeat: fix Mojarra ui:repeat does not reset EL var correctly

### ISSUE THREAD

**Issue by rabiter** [primefaces/primefaces#6964]
**Describe the defect**
The "resetValues=true" attribute + UIRepeat behave not as expected (please check the attached application and the code snippet below)

**Reproducer**
primefaces-test application is attached
[primefaces-test-master.zip](https://github.com/primefaces/primefaces/files/5946677/primefaces-test-master.zip)

**Environment:**
 - PF Version: 10.0.0
 - JSF + version: Mojarra (all versions)
 - Affected browsers: ALL_

**To Reproduce**
1. Go to the index page on attached primefaces-test application
2. Click on 'Reproduce bug' link

**Expected behavior**
I expect the variable #{value} stay empty

**Example XHTML**
```html
    <h:form>
        Expected to stay empty: #{value}
    </h:form>
    <h:form>
        <ui:repeat value="#{testView.strings}" var="value">
            <p:outputLabel value="#{value} "/>
        </ui:repeat>
    </h:form>

    <h:form>
        <!-- this link will update the variable value in the first form with value from the UIRepeat -->
        <p:commandLink value="Reproduce bug" process="@this" update="@(form)" resetValues="true"/>
        <!-- and this link works fine -->
        <p:commandLink value="Not reproduce bug" process="@this" update="@(form)" resetValues="false"/>
    </h:form>
```

**Example Bean**
```java
@Data
@Named
@ViewScoped
public class TestView implements Serializable {
    
    private final List<String> strings = new ArrayList<>();

    @PostConstruct  
    public void init() {
        strings.addAll(Arrays.asList("Winona Lindsey", "Pandora Peay", "Honey Dean"));
    }

    public List<String> getStrings() {
        return strings;
    }
}
```

**Comment by tandraschko** [comment]
Its mojarra bug. Seems like the var of the ui:repeat is not removed from EL.
So its also reproduceable in p:repeat, which is a fork of mojarra, to fix a old bug in it.

You should at least create a mojarra issue. We could also fix it in p:repeat but i would actually like to get rid of this hack.

**Comment by tandraschko** [comment]
as its not a PrimeFaces bug, i will set it as enhancement. Would be great if you work on a PR for Mojarras ui:repeat and port the fix to p:repeat

**Comment by tandraschko** [comment]
Can can simple debug p:repeat, line 671:
If visitTree is called (with PrimePartialViewContext#resetValues in stack), the following results in null:
context.getFacesContext().getExternalContext().getRequestMap().get(this.var)

when you also add a breakpoint in ln727, the var isnt removed and it resolves in: Winona Lindsey

maybe @melloware would like to fix it in p:repeat and create a mojarra issue+PR?

**Comment by melloware** [comment]
I will put this on my TODO list.  I am most dissapointed with Mojarra progress since switching from Oracle to Jakarta.  I have had two open PR's for a long time that fix simple and obvious bugs and just absolutely no activity.

**Comment by rabiter** [comment]
the Mojarra UIRepeat issue seems to be at least 8 years old:
https://stackoverflow.com/questions/14398960/magic-uirepeat-var

**Comment by melloware** [comment]
Reported to Mojarra and provided a PR.

Issue: https://github.com/eclipse-ee4j/mojarra/issues/4830

PR: https://github.com/eclipse-ee4j/mojarra/pull/4831

**Comment by arjantijms** [comment]
> I am most dissapointed with Mojarra progress since switching from Oracle to Jakarta. 

I am actually somewhat disappointed myself too :( From being one of the many contributors to Mojarra, I'm suddenly pretty much the only one. If you want I can nominate you though to become committer, so it's not essentially me who's the bottleneck. Let me know if interested.

Btw, on the issue of resetting EL vars in general, almost a decade ago I also created issues for this, but with Mojarra and with PrimeFaces. If I remember correctly Cagatay responded that he would mediate over it. The idea I proposed was creating a kind of internal "scope" (not a true scope), which would clear out EL variables local to a component automatically.
